File: 덕현/2026-05/21_pgr_택배상자.py
'''
프로그래머스 131704 택배상자

입력:
order

구조:
stack으로해서 answer+1
'''

def solution(order):
    answer = 0
    stack = []

    for num in range(1, len(order) + 1):
        stack.append(num)

        while stack:
            top = stack[-1]

            if top == order[answer]:
                stack.pop()
                answer += 1
            else:
                break

            if answer == len(order):
                break

    return answer

File: su/2026-05/21_pgr_택배상자.java
class Solution {
    public int solution(int n, int w, int num) {
        int row = (num - 1) / w;
        int col = (row % 2 == 0) ? (num - 1) % w : (w - 1 - (num - 1) % w);
        int lastRow = (n + w - 1) / w;
        int answer = 0;

        for (int r = row; r < lastRow; r++) {
            int box = (r % 2 == 0) ? r * w + col + 1 : r * w + (w - col);
            if (box <= n) answer++;
        }

        return answer;
    }
}